File: sigma_marketing_api/SigmaMarketing/SigmaMarketing.Model/Models/Request/CampaignAnalyticsRequest.cs
namespace SigmaMarketing.Model.Models.Request
{
    public class CampaignAnalyticsRequest
    {
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public List<int>? Statuses { get; set; } = null;
        public List<int>? Platforms { get; set; } = null;
        public List<int>? Tags { get; set; } = null;
        public int Page { get; set; }
        public int PageSize { get; set; }
    }
}

File: sigma_marketing_app/lib/data/models/request/brand/campaign_analytics/campaign_analytics_request.dart
class CampaignAnalyticsRequest {
  DateTime startDate;
  DateTime endDate;
  List<int>? statuses;
  List<int>? platforms;
  List<int>? tags;
  int page;
  int pageSize;

  CampaignAnalyticsRequest({
    required this.startDate,
    required this.endDate,
    this.statuses,
    this.platforms,
    this.tags,
    required this.page,
    required this.pageSize,
  });

  Map<String, dynamic> toJson() {
    return {
      'startDate': startDate.toIso8601String(),
      'endDate': endDate.toIso8601String(),
      'statuses': statuses,
      'platforms': platforms,
      'tags': tags,
      'page': page,
      'pageSize': pageSize,
    };
  }
}
